data/data/alibabacloud/cluster/dns/privatezone.tf

```diff
@@ -15,7 +16,7 @@ data "alicloud_alidns_domains" "dns_public" {
 }
 
 resource "alicloud_alidns_record" "dns_public_record" {
-  count = length(data.alicloud_alidns_domains.dns_public.domains) == 0 ? 0 : 1
+  count = local.is_external && length(data.alicloud_alidns_domains.dns_public.domains) != 0 ? 1 : 0
```

Comment:
As per https://bugzilla.redhat.com/show_bug.cgi?id=2041926, we actually want this to fail when `local.is_external` is true and `length(data.alicloud_alidns_domains.dns_public.domains` is 0. In that case, the user requested external publishing but supplied a base domain for which there is no zone. Nevertheless, we can keep this as is here and address the BZ in a separate issue.